src/InstrumentationEngine.Attach/CommandHandler/AttachCommandHandler.cs

```diff
@@ -0,0 +1,513 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Diagnostics.NETCore.Client;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Xml;
+using System.Xml.Serialization;
+using static System.FormattableString;
+
+namespace Microsoft.InstrumentationEngine
+{
+    /// <summary>
+    /// Intermediate class used to track ConfigurationSources when generating an EngineConfiguration.
+    /// </summary>
+    internal class ConfigurationSourceInfo
+    {
+        public InstrumentationConfigurationSources? Sources { get; set; }
+
+        public string? ConfigSourceDirectory { get; set; }
+
+        public string? ConfigSourceFilePath { get; set; }
+    }
+
+    [Flags]
+    internal enum LoggingFlags
+    {
+        None = 0,
+        Errors = 1,
+        Messages = 2,
+        Dumps = 4
+    }
+
+    internal class AttachCommandHandler
```

Comment:
>AttachCommandHandler [](start = 19, length = 20)

Any significant changes in the body here or is it mostly a refactor?

Reply:
Mostly a refactor, but also added multiple configs and engine logging parameters

---
In reply to: [394138338](https://github.com/microsoft/CLRInstrumentationEngine/pull/234#discussion_r394138338) [](ancestors = 394138338)